src/routes/contracts/__tests__/get-contract.e2e-spec.ts

```diff
@@ -0,0 +1,63 @@
+import * as request from 'supertest';
+import { INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { AppModule } from '../../../app.module';
+import { readFileSync } from 'fs';
+import { Contract } from '../entities/contract.entity';
+import { RedisClientType, createClient } from 'redis';
+
+describe('Get contract e2e test', () => {
+  let app: INestApplication;
+  let redisClient: RedisClientType;
+  const chainId = '5'; // Görli testnet
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleRef.createNestApplication();
+    await app.init();
+
+    redisClient = await redisClientFactory();
+    await redisClient.flushAll();
```

Comment:
In this case it doesn't matter so much since there's only a test (and it won't be more tests for /contracts I think), but I agree that _in the majority cases_ is a best practice to flush per test. I will send a PR to change it 🙂

Reply:
In this case that is true I think it's better to avoid a situation where we are debugging the state of one test only to find out that the state is reset only after the suite finishes.

So even though both accomplish the same thing I still think `beforeEach` and `afterEach` are more correct and represent better what needs to happen at the beginning, before each test, after each test and at the end.

Reply:
Changed [here](https://github.com/5afe/safe-client-gateway-nest/pull/145) 👍